analyzers/tests/SonarAnalyzer.UnitTest/Extensions/AttributeDataExtensionsTest.cs

```diff
@@ -62,6 +63,95 @@ public class AttributeDataExtensionsTest
         public void HasAnyNameThrowsForNull() =>
             new Action(() => AttributeDataWithName("TestAttribute").HasAnyName(null)).Should().Throw<Exception>();
 
+        [DataTestMethod]
+        [DataRow("SomeBool", typeof(bool), true, true)]
+        [DataRow("someBool", typeof(bool), true, true)]
+        [DataRow("somebool", typeof(bool), true, true)]
+        [DataRow("SOMEBOOL", typeof(bool), true, true)]
+        [DataRow("SomeInt", typeof(int), true, 1_234_567)]
+        [DataRow("SomeInt", typeof(byte), false, (byte)0)]
+        [DataRow("SomeByte", typeof(byte), true, (byte)24)]
+        [DataRow("SomeByte", typeof(int), true, 24)]
+        [DataRow("SomeString", typeof(string), true, "Text")]
+        [DataRow("SomeNull", typeof(string), true, null)]
+        [DataRow("Missing", typeof(string), false, null)]
+        [DataRow("SomeString", typeof(int), false, 0)]
+        [DataRow("SomeNumberString", typeof(int), true, 42)]
+        public void TryGetAttributeValue_Arguments(string valueName, Type valueType, bool expectedSucess, object expectedResult)
+        {
+            var arguments = new Dictionary<string, object>
+            {
+                { "SomeBool", true },
+                { "SomeInt", 1_234_567 },
+                { "SomeByte", (byte)24 },
+                { "SomeString", "Text" },
+                { "SomeNumberString", "42" },
+                { "SomeNull", null },
+            };
+            TryGetAttributeValue_Arguments(valueName, valueType, expectedSucess, expectedResult, namedArguments: arguments);       // [Attr(SomeBool = true, SomeInt = 1_234_567, ..)]
+            TryGetAttributeValue_Arguments(valueName, valueType, expectedSucess, expectedResult, constructorArguments: arguments); // [Attr(SomeBool: true, SomeInt: 1_234_567, ..)]
+
+            static void TryGetAttributeValue_Arguments(string valueName,
```

Comment:
Remove the static. That will allow to remove everything that doesn't need to be here